salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+                    new_auth_methods.append(auth_method.type)
+
+                # Provision config for specific auth method
+                if auth_method.auth_config:
+                    if auth_method.type == "ldap":
+                        log.debug('Provisioning configuration for LDAP...')
+                        client.auth.ldap.configure(**auth_method.auth_config)
+                        log.debug('Configuration for LDAP is provisioned.')
+                else:
+                    log.debug(
+                        'Auth method "%s" does not contain any specific configurations.', auth_method.type)
+
+                if auth_method.extra_config:
+                    log.debug(
+                        'Provisioning extra configurations for auth method "%s"', auth_method.type)
+                    # Get LDAP group mapping from vault
+                    ldap_list_group_response = client.auth.ldap.list_groups()
+                    if ldap_list_group_response:
+                        ldap_groups = ldap_list_group_response["data"]["keys"]
+
+                    log.debug("LDAP groups from vault: %s", str(ldap_groups))
+
+                    # Update LDAP group mapping
+                    log.debug(
+                        'Writing LDAP group -> Policy mappings for "%s"', str(auth_method.path))
+                    local_config_groups = auth_method.extra_config["group_policy_map"]
+                    for key in local_config_groups:
+                        log.debug('LDAP Group ["%s"] -> Policies %s',
+                                  str(key), local_config_groups[key])
+
+                        client.auth.ldap.create_or_update_group(
+                            name=key,
+                            policies=local_config_groups[key]
+                        )
+
+                    # Clean up LDAP group mapping
+                    if ldap_groups:
+                        for group in ldap_groups:
+                            if group in {k.lower(): v for k, v in local_config_groups.items()}:
+                                log.debug(
+                                    'LDAP group mapping ["%s"] exists in configuration, no cleanup necessary', group)
+                            else:
+                                log.debug(
+                                    'LDAP group mapping ["%s"] does not exist in configuration, deleting...', group)
+                                client.auth.ldap.delete_group(name=group)
+                                log.debug(
+                                    'LDAP group mapping ["%s"] deleted.', group)
+                else:
+                    log.debug(
+                        'Auth method "%s" does not contain any extra configurations.', auth_method.type
+                    )
+
+            # Build return object
+            ret['changes']['old'] =[ob.type for ob in remote_methods]
+            ret['changes']['new'] = new_auth_methods or "No changes"
+
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.